lib/TextDocument/LocationRanges.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Phpactor\TextDocument;
+
+use ArrayIterator;
+use Countable;
+use IteratorAggregate;
+use RuntimeException;
+
+/**
+ * @implements IteratorAggregate<LocationRange>
+ */
+final class LocationRanges implements IteratorAggregate, Countable
```

Comment:
as with the above, I doubt we need this. the `Locations` collection should be sufficient